lib/api/model/pokemon_ability.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'pokemon_ability.freezed.dart';
+part 'pokemon_ability.g.dart';
+
+@freezed
+class PokemonAbility with _$PokemonAbility {
```

Comment:
no need to add the prefix Pokemon here, Ability is enough 🙂 

apply same thing for other data classes except for `pokemon_details.dart` just leave it as is 🙂